webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/AgglomerateService.scala

```diff
@@ -130,77 +135,73 @@ class AgglomerateService @Inject()(config: DataStoreConfig) extends DataConverte
+            position = Point3D(pos(0).toInt, pos(1).toInt, pos(2).toInt)
+          )
+      }
 
-    val hdfFile =
-      dataBaseDir
-        .resolve(organizationName)
-        .resolve(dataSetName)
-        .resolve(dataLayerName)
-        .resolve(agglomerateDir)
-        .resolve(s"${mappingName}.${agglomerateFileExtension}")
-        .toFile
+      val skeletonEdges = edges.map { e =>
+        Edge(source = e(0).toInt, target = e(1).toInt)
+      }
 
-    val reader = HDF5FactoryProvider.get.openForReading(hdfFile)
-    val positionsRange: Array[Long] =
-      reader.uint64().readArrayBlockWithOffset("/agglomerate_to_segments_offsets", 2, agglomerateId)
-    val edgesRange: Array[Long] =
-      reader.uint64().readArrayBlockWithOffset("/agglomerate_to_edges_offsets", 2, agglomerateId)
-
-    val nodeCount = positionsRange(1) - positionsRange(0)
-    val edgeCount = edgesRange(1) - edgesRange(0)
-    if (nodeCount > config.Braingames.Binary.agglomerateSkeletonEdgeLimit) {
-      throw new Exception(
-        s"Agglomerate has too many nodes ($nodeCount > ${config.Braingames.Binary.agglomerateSkeletonEdgeLimit}")
-    }
-    if (edgeCount > config.Braingames.Binary.agglomerateSkeletonEdgeLimit) {
-      throw new Exception(
-        s"Agglomerate has too many edges ($edgeCount > ${config.Braingames.Binary.agglomerateSkeletonEdgeLimit}")
-    }
-    val positions: Array[Array[Long]] =
-      reader.uint64().readMatrixBlockWithOffset("/agglomerate_to_positions", nodeCount.toInt, 3, positionsRange(0), 0)
-    val edges: Array[Array[Long]] =
-      reader.uint64().readMatrixBlockWithOffset("/agglomerate_to_edges", edgeCount.toInt, 2, edgesRange(0), 0)
-
-    val nodes = positions.zipWithIndex.map {
-      case (pos, idx) =>
-        Node(
-          id = idx,
-          position = Point3D(pos(0).toInt, pos(1).toInt, pos(2).toInt),
-          rotation = Vector3D(0, 0, 0),
-          viewport = 0,
-          resolution = 1,
-          bitDepth = 0,
-          interpolation = false,
-          radius = 120,
-          createdTimestamp = System.currentTimeMillis()
-        )
-    }
+      val trees = Seq(
+        Tree(
+          treeId = agglomerateId.toInt,
+          createdTimestamp = System.currentTimeMillis(),
+          nodes = nodes,
+          edges = skeletonEdges,
+          name = s"agglomerate ${agglomerateId} (${mappingName})"
+        ))
+
+      val skeleton = SkeletonTracingDefaults.createInstance.copy(
+        dataSetName = dataSetName,
+        trees = trees,
+        organizationName = Some(organizationName)
+      )
+      val duration = System.nanoTime() - startTime
+      if (duration > 100 * 1e6) {
+        logger.debug(s"Generating skeleton from agglomerate file took ${math
+          .round(duration / 1e6)} ms (${skeletonEdges.length} edges, ${nodes.length} nodes).")
+      }
 
-    val skeletonEdges = edges.map { e =>
-      Edge(source = e(0).toInt, target = e(1).toInt)
+      Fox.successful(skeleton)
```

Comment:
I'd prefer a simple box here, but that is personal preference, so leave it as it is.